libclc/utils/libclc-remangler/LibclcRemangler.cpp

```diff
@@ -147,340 +123,606 @@ class BumpPointerAllocator {
   }
 
   ~BumpPointerAllocator() { reset(); }
+
+private:
+  void grow() {
+    char *NewMeta = static_cast<char *>(std::malloc(AllocSize));
+    if (NewMeta == nullptr)
+      std::terminate();
+    BlockList = new (NewMeta) BlockMeta{BlockList, 0};
+  }
+
+  void *allocateMassive(size_t NBytes) {
+    NBytes += sizeof(BlockMeta);
+    BlockMeta *NewMeta = reinterpret_cast<BlockMeta *>(std::malloc(NBytes));
+    if (NewMeta == nullptr)
+      std::terminate();
+    BlockList->Next = new (NewMeta) BlockMeta{BlockList->Next, 0};

```

Comment:
@AaronBallman, does this statement have sequencing issues or is it well sequenced?

Reply:
Okay, so I am glad I was not the only person who looked at that code and went "is that right?"! :-D

There's no sequencing problem there because of: https://eel.is/c++draft/expr.ass#1